File: 1127-last-stone-weight/last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq=new PriorityQueue<>(Collections.reverseOrder());
        for(int s:stones) pq.add(s);
        while(pq.size()>1){
            int mx1=pq.remove();
            int mx2=pq.remove();
            if(mx1 !=mx2){
                pq.add(mx1-mx2);

            }
        }
        if(pq.isEmpty()) return 0;
        return pq.peek();
    }
}

File: 1127-last-stone-weight/last-stone-weight.py
class Solution(object):
    def lastStoneWeight(self, stones):
        s=stones
        while len(s)>1:
            s.sort(reverse=True)
            q=s.pop(0)
            p=s.pop(0)
            if q!=p:
                s.append(q-p)
        if s:
            return s[0]
        return 0


